### ISSUE TITLE
drop in mAP after synthesis

### ISSUE THREAD

**Issue by nikky4D** [MaximIntegratedAI/ai8x-training#259]
I am training a one channel model with the fpn_detector and I notice a drop in accuracy in two parts. In the training, after the start of quantization (at epoch 400), I notice a drop in the mAP values. These values recover to some extent and continue training ending at about 0.68 mAP on train, with 0.64 mAP on the test set. Do you have some idea as to why that happens? 

![image](https://github.com/MaximIntegratedAI/ai8x-training/assets/7451106/4e46e625-3a88-4203-bc6b-49586aeb6ca8)

However, after training is finished, I also see that after synthesis, my accuracy on my test set drops dramatically, for instance, a drop of 3-4 points from 0.6 - 0.7 to about 0.2-0.3mAP. I haven't seen this type of drop so I think it may be something in my training. I utilize ai8x.normalize in my dataloader, but still see this drop. Is there anything you recommend to check for to improve the synthesis results?

**Comment by MaximGorkem** [comment]
Hi, 

Firstly, it is normal to experience an accuracy drop after QAT starts. With the QAT, the weights are clamped and quantized and the activations are quantized in the forward operation. The aim of QAT is to continue the training with these effects so that the parameters of the network compensate for the effects of quantization.

For the accuracy drop for the test case: The checkpoint file that is produced with the QAT process is still in FP32, since FP values are still needed in the backpropagation. However, the weights and biases are optimized with the presence of quantization effects. After the training, we are feeding this checkpoint to quantize.py to produce integer model from the FP model. I assume you are testing this quantized model with your test set and seeing some accuracy drop. Initially, it would be logical to verify that this difference is not caused by the discrepancies between your validation and test sets. Have you tried evaluating the validation set with the quantized model? You should see the same performance (the performance you observe in the training step where you create the FP checkpoint).
